addons/project/views/project_views.xml

```diff
@@ -71,7 +71,8 @@
                         <filter string="Stage" name="stage" context="{'group_by': 'stage_id'}"/>
                         <filter string="Personal Stage" name="personal_stage" context="{'group_by': 'personal_stage_type_ids'}"/>
                         <filter string="Assignees" name="user" context="{'group_by': 'user_ids'}"/>
-                        <filter string="Project" name="project" context="{'group_by': 'project_id'}"/>
+                        <filter string="Project" name="project" context="{'group_by': 'project_id'}" invisible="context.get('default_project_id', False)"/>
```

Comment:
I would rather like to create a primary inherited search view that would be used in "My Tasks" where the display_project_id field would be added in the "Group By".
This is a bit too hackish for me.

After having read the diff a bit more, shouldn't we just move it into `view_task_search_form_extended`? Isn't it the right one? I haven't tested though, could you check?